KeeTrayTOTP/Menu/MenuItemProvider.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.Windows.Forms;
+using KeePass.Plugins;
+
+namespace KeeTrayTOTP.Menu
+{
+    public class MenuItemProvider : IDisposable
+    {
+        private readonly MenuItemProviderBase _trayMenuItemProvider;
+        private readonly MenuItemProviderBase _entryMenuItemProvider;
+        private readonly MenuItemProviderBase _mainMenuItemProvider;
+
+        public MenuItemProvider(KeeTrayTOTPExt plugin, IPluginHost pluginHost)
+        {
+            if (pluginHost.CustomConfig.GetBool(KeeTrayTOTPExt.setname_bool_LegacyTrayMenuProvider_Enable, false))
+            {
+                _trayMenuItemProvider = new LegacyTrayMenuItemProvider(plugin, pluginHost);
+            }
+            else
+            {
+                _trayMenuItemProvider = new TrayMenuItemProvider(plugin, pluginHost);
+            }
+
+            _entryMenuItemProvider = new EntryMenuItemProvider(plugin, pluginHost);
+            _mainMenuItemProvider = new MainMenuItemProvider(plugin);
+        }
+
+        internal ToolStripMenuItem GetMenuItem(PluginMenuType type)
+        {
+            switch (type)
```

Comment:
Nice, way cleaner.

Reply:
Btw, let me know once you're done with the draft. Happy to review :)

Reply:
Currently adding more unit tests and fixing bugs.